Branch-SDK/src/io/branch/referral/PrefHelper.java

```diff
@@ -332,10 +333,22 @@ public String getAppKey() {
     }
 
 	public void setBranchKey(String key) {
-	    Branch_Key = key;
-    }
+		Branch_Key = key;
+		String currentBranchKey = getString(KEY_BRANCH_KEY);
+		if (key == null || currentBranchKey == null || !currentBranchKey.equals(key)) {
```

Comment:
`currentBranchKey` here is the last saved branch key.Any time there is a  new  branch key it is saved to the preference. Is there any thing missing?